src/diff.c

```diff
@@ -83,8 +83,6 @@ static bool
 diff_common_add_cell(struct diff_stat_context *context, size_t length)
 {
 	assert(ARRAY_SIZE(context->cell) > context->cells);
-	if (length == 0)
-		return true;
```

Comment:
Would be great to only allow empty cells at the end of the line. Need to take a closer look.